### ISSUE TITLE
FragmentTree + Hider

### ISSUE THREAD

**Issue by MarcGM15** [IFCjs/components#195]
### Description 📝

As a Bim engineer using IFC.js it would be nice to have the hierarchy tree of the ifc with the option to hide or show the elements like there is in the hider element. In this way with one feature we have the two funcionalities

### Suggested solution 💡

In the fragmentTree give the possibility to have a checkbox with the element name to manage the visibility of this element

### Validations ✅

- [X] Read the [docs](https://platform.thatopen.com/documentation).
- [X] Check that there isn't [already an issue](https://github.com/IFCjs/components/issues) that requests the same feature to avoid creating a duplicate.

**Comment by HoyosJuan** [comment]
Hi @MarcGM15!

I would like to think in a more customizable way for the end user to decide what to do with the elements related in each row of the hierarchy. What about adding the possibility to set action icons / a drop down menu at the right side in order for developers to add any buttons they like and perform any kind of functionalities? 

That way, you would be able to add a button to the right side in order to show or hide the elements contained in that hierarchy level.

What do you think about it? Does it suit better the needs?

**Comment by MarcGM15** [comment]
I think it's great! Thank you

**Comment by daniel-s-goncalves** [comment]
Until this gets added, I just wanna say that you can generate a spatial tree of the IFC with some workarounds. Unfortunately, the load() method of the FragmentIfcLoader class cleans up the instantiated WEBIFC.IfcAPI object (stored in the local variable this._webIfc), which would allow you to acquire the properties through WEBIFC.Properties and then call the getSpatialStructure() method. 

To get around this, you can either: manually load the model again using the WEBIFC.IfcAPI **OR** make a copy of the repository and locally modify the load() function to return a spatial tree and prevent having to load the entire IFC again.

Inside the load() function of the FragmentIfcLoader, the code would look like this:

```
// Right after the await this.readIfcFile() method of the load() method in the FragmentIfcLoader class:

const prop = new WEBIFC.Properties(this._webIfc)
const tree = await prop.getSpatialStructure(modelID)
```

The `tree` will be of the type "Node", which complies with the following interface:

```
interface Node {
    expressID: number;
    type: string;
    children: Node[];
}
```

You can then display a tree-like structure, using an external library. If you are using React, like [react-accessible-treeview](https://www.npmjs.com/package/react-accessible-treeview).

**Comment by HoyosJuan** [comment]
Hey @daniel-s-goncalves! Thanks for your message.

Please, correct me if I'm wrong, but Marc is not talking about the possibility to create a tree, but to make the existing one more customizable/extensible. At the time being, no workaround is needed to create hierarchies. You can achieve them by using the FragmentClassifier and FragmentTree 🙂 without the need to depend on the instance of web-ifc.

**Comment by agviegas** [comment]
Solved by #211 ! Will be available in the next beta release, or in the next milestone release.
